src/repl/scala/tools/nsc/interpreter/PresentationCompilation.scala

```diff
@@ -136,40 +136,57 @@ trait PresentationCompilation { self: IMain =>
       typeString(typedTreeAt(start, end))
 
     val NoCandidates = (-1, Nil)
-    type Candidates = (Int, List[String])
+    type Candidates = (Int, List[CompletionCandidate])
 
-    override def candidates(tabCount: Int): Candidates = {
+    override def completionCandidates(tabCount: Int): Candidates = {
       import compiler._
       import CompletionResult.NoResults
 
+      def isMemberDeprecated(m: Member) = m match {
+        case tm: TypeMember if tm.viaView.isDeprecated || tm.viaView != NoSymbol && tm.viaView.owner.isDeprecated || !m.sym.exists =>
+          true
+        case _ =>
+          m.sym.isDeprecated ||
+            m.sym.hasGetter && m.sym.getterIn(m.sym.owner).isDeprecated
+      }
+      def isMemberUniversal(m: Member) =
+        definitions.isUniversalMember(m.sym) ||
+          (m match {
+            case t: TypeMember =>
+              t.implicitlyAdded && t.viaView.info.params.head.info.bounds.isEmptyBounds
+            case _ =>
+              false
+          })
+      def memberArity(m: Member): CompletionCandidate.Arity =
+        if (m.sym.paramss.isEmpty) CompletionCandidate.Nullary
+        else if (m.sym.paramss.size == 1 && m.sym.paramss.head.isEmpty) CompletionCandidate.Nilary
+        else CompletionCandidate.Other
```

Comment:
Now I'll never forget which is which.

Reply:
Why isn't a function in two params called "diary"?

Reply:
Why isn't an interface that takes only bees called "apiary"?

Similarly "aviary" for the windows format that's for the birds.

Reply:
teach me about the birds and the bees